src/ConstructiveSolidGeometry/VolumePrimitives/RegularPrism.jl

```diff
@@ -19,19 +19,34 @@ const HexagonalPrism{T,CO,TR}  = RegularPrism{T,CO,6,TR}
 function Geometry(::Type{T}, ::Type{P}, dict::AbstractDict, input_units::NamedTuple, transformations::Transformations{T}
             ) where {T, P <: Union{TriangularPrism, QuadranglePrism, PentagonalPrism, HexagonalPrism}}
     length_unit = input_units.length
+    angle_unit = input_units.angle
+    origin = get_origin(T, dict, length_unit)
+    rotation = get_rotation(T, dict, angle_unit)
+
     r = parse_r_of_primitive(T, dict, length_unit)
-    z = parse_height_of_primitive(T, dict, length_unit)
-    hZ = typeof(z) <: Real ? z : width(z)/2
-    origin = if typeof(z) <: Real
-        CartesianPoint{T}(zero(T), zero(T), zero(T))
-    else
-        CartesianPoint{T}(zero(T), zero(T), mean(z))
+    hZ = if haskey(dict, "h")
+        _parse_value(T, dict["h"], length_unit) / 2
+    elseif haskey(dict, "z")
+        z = parse_height_of_primitive(T, dict, length_unit)
+        hZ = typeof(z) <: Real ? z : (z[2] - z[1])/2
```

Comment:
For consistency: `z isa Real` like the line below

Reply:
Ok.